src/CMakeLists.txt

```diff
@@ -253,80 +247,80 @@ if(WITH_LZ4_EXT)
   target_link_libraries(rdkafka PUBLIC LZ4::LZ4)
 endif()
 
-# Set up path to these sources for other sub-projects (tests, examples)
-# to be able to reach them.
-#set(rdkafka_SRC_DIR ${PROJECT_SOURCE_DIR}
-#    CACHE INTERNAL "${PROJECT_NAME} source dir" FORCE)
+if(MINGW)
+  message(STATUS "Detected MinGW build")
+  # MinGW ignores `#pragma comment(lib, ...)` so we need to specify link libs
+  target_link_libraries(rdkafka PUBLIC crypt32 ws2_32 secur32)
+endif(MINGW)
 
 # Generate pkg-config file
-set(PKG_CONFIG_NAME
-    "librdkafka"
-)
-set(PKG_CONFIG_DESCRIPTION
-    "The Apache Kafka C/C++ library"
-)
-set(PKG_CONFIG_VERSION
-    "${PROJECT_VERSION}"
-)
+set(PKG_CONFIG_VERSION "${PROJECT_VERSION}")
 set(PKG_CONFIG_REQUIRES "")
-if(WITH_ZLIB)
-  string(APPEND PKG_CONFIG_REQUIRES "zlib ")
+if (WIN32)
+  set(PKG_CONFIG_LIBS_PRIVATE "-lws2_32 -lsecur32 -lcrypt32")
+else()
+  set(PKG_CONFIG_LIBS_PRIVATE "-lpthread")
+  find_library(RT_LIBRARY rt)
+  if(RT_LIBRARY)
+    string(APPEND PKG_CONFIG_LIBS_PRIVATE " -lrt")
+  endif()
+
+  if(WITH_PLUGINS)
```

Comment:
WITH_PLUGINS is satisfied by Win32 (LoadLibrary) or libdl, so this should probably be WITH_LIBDL (but use LoadLibrary, not libdl on win32)